gateway/ln-gateway/src/lib.rs

```diff
@@ -81,20 +81,19 @@ impl LnGateway {
     }
 
     async fn load_federation_actors(&self) {
-        // TODO: handle error better
-        for config in self
-            .client_builder
-            .load_configs()
-            .expect("Could not read configs")
-        {
-            let client = self
-                .client_builder
-                .build(config.clone())
-                .expect("Could not build federation client");
-
-            if let Err(e) = self.register_federation(Arc::new(client)).await {
-                error!("Failed to register federation: {}", e);
+        if let Ok(configs) = self.client_builder.load_configs() {
+            for config in configs {
+                let client = self
+                    .client_builder
+                    .build(config.clone())
+                    .expect("Could not build federation client");
```

Comment:
What could trigger this btw?

Reply:
A mis-configured client builder might fail to load any previous federation configs, or might fail to parse the loaded config(s).
It's a small chance but this might happen on manual restarts of gateway, especially after upgrades